### ISSUE TITLE
Question 4: What OS and software need to be installed?

### ISSUE THREAD

**Issue by Emersont1** [Emersont1/TechNottingham.IT-Refurb#4]
Peter Taylor  1 hour ago
Probably something Linux based would be best because we should be able to automate it too (links in with #5)

mpastuszka  1 hour ago
Depends on the end users likely to use them - are they likely to require Windows?

c.se:party-wizard:  1 hour ago
I think that this depends on the device and how new the OS is, i.e. if it's running an old version of Windows and can't be free-upgraded then just nope on buying new Windows 10 licenses.
But we need to consider that some recipients may be reticent to use Linux systems so I think trying to keep options open would be good.

c.se:party-wizard:  1 hour ago
Also we may receive Mac OS and Chromebook devices. These need to keep their original OS. As well, some Windows laptops play nastily with Linux (Surfaces can be tetchy last I recall) so I don't know if this can be one size fits all. (edited) 

c.se:party-wizard:  44 minutes ago
(but if we can identify what the different situations are, then it will probably work for some of them!)

mpastuszka  25 minutes ago
with regards to software, OpenOffice/LibreOffice or something along those lines would be a good shout.
:+1:
1

**Comment by cruxicheiros** [comment]
I think that if we are in the situation of having old devices with old OSes and need to replace whatever they're originally running, we should consider [Xubuntu](https://xubuntu.org/) because XFCE is lightweight and it could come with a speed increase. That said, a lot of internet guides for Ubuntu expect the default desktop environment, so it could be better to go for just plain Ubuntu.

In any case, Ubuntu works on the most stuff out of the box and is well documented. I think that if our main motivation is delivering computers that work and which don't cause upkeep problems for end users, we should be mainly considering shipping Ubuntu when it comes to the question of 'which Linux'.

**Comment by Emersont1** [comment]
Possibly ubuntu or mint, but maybe is a rolling release based distro a better idea so you're not having issues with old packages fixed a long time ago?

**Comment by cruxicheiros** [comment]
That is a good idea, and we should probably be installing LTS versions.

**Comment by cruxicheiros** [comment]
(have just realised that LTS precludes rolling-release, but I think it would be a more stable user experience and also mean that things like security updates would be available for longer - we can't necessarily provide support if a rolling release breaks something.)

**Comment by cruxicheiros** [comment]
If someone is really keen to install a particular distro on a computer they get from us then they can do that themselves after they get it, but I think the safest assumption is that change is not a desirable thing.

**Comment by Emersont1** [comment]
in about 4 years of arch, i've only had one gnupg update break stuff that may have been saved by having something not be RR based, so imo thats less of an issue than people make it out to be

**Comment by cruxicheiros** [comment]
At the risk of further stoking a distro war: it may cause ire to linux purists, but Ubuntu's proprietary driver support out of the box is very good and means it's more likely to work immediately with arbitrary machines. We are also doing this project to send machines to an arbitrary set of people who may have differing amounts of tech savviness. Someone who wants to use Arch (or other RR distros) is more likely to be able to comfortably install one than someone who has one installed but would be better off using Ubuntu - that's why I prefer it for this situation.

**Comment by monotron** [comment]
I'm with @cruxicheiros on this. If we're going to be posting these out to people to use, we want to make sure that:

- OS installation isn't a huge pain (Arch can take forever to set up and requires specific knowledge of how to do it)
- End users don't have to fight issues when the system gets updated/there's a hardware change

Therefore I think a more "stable" distro like Ubuntu would be a better choice in this respect, especially since there are lightweight versions (Lubuntu/Xubuntu/Ubuntu MATE) we could just as easily use on older machines.

As for machines that would be fine on Windows, **generally speaking** any machine that has been activated before on Windows 10, or which has an [SLP key](https://en.wikipedia.org/wiki/System_Locked_Pre-installation) built into the BIOS for an older version of Windows, **should** activate just fine under Windows 10 (I don't have any hardware laying around to test this more definitively and figure out what the exceptions are, but this has always been my experience when reinstalling OSes for people).

So IMHO I think the plan should be as follows:
- Establish a baseline for what system specs are required to deliver a reasonable experience under Windows 10.
    - If the system meets these requirements, attempt to install Windows 10. If we run into activation/licensing issues we can fall back to a Linux-based OS.
    - If the system does not meet these requirements, install Linux instead.

I'd love to hear your folks' feedback on what I've put forward here, as I have undoubtedly missed something important.

**Comment by markpastuszka** [comment]
Yes, generally anything with an 8 or 8.1 SLP key in the BIOS will activate Windows 10 - it should pick up the right version in the installer without ever having to enter a key.

Machines originally shipping with Windows 7 may activate a 10 install when connected to the internet if they have previously been upgraded to 10/had 10 installed on them. Windows 7 license keys can be used to activate Windows 10 as well.

I assume that we are unlikely to get anything pre-Windows 7, but for these I'd say it would be easier to go with Linux (they would probably be under the aforementioned baseline spec as well)
Rough suggestions for a baseline spec:
Minimum 3GB RAM (2GB is pushing it)
Dual core
Came with Windows 7 or newer

**Comment by cruxicheiros** [comment]
@monotron: Adding to that, @markpastuszka on Slack brought up the fact that a lot of Windows machines have a product key printed on a sticker where if the sticker is still present and legible, you can use it to install Windows 10 using a tool. Win 7 product keys can also be used to get the free upgrade - [here's documentation on how](https://www.zdnet.com/article/heres-how-you-can-still-get-a-free-windows-10-upgrade/).

@markpastuszka: as a sort of jokey response: you can get all sorts of machines that should never-ever have had Windows installed on them, 2gb netbooks etc. I think if those pop up, you can just put Linux on them. It should be noted that if they come from past a certain point, you need to `jump through a through hoops before doing that, though.](https://www.pcworld.com/article/2951559/how-to-install-linux-on-a-pc-with-secure-boot-enabled.html)

**Comment by monotron** [comment]
I'd be inclined to agree with @markpastuszka here on the baseline spec, with the caveat that laptops with Intel Atom CPUs are best to install Linux on regardless of core count. Unfortunately Windows 10 dropped support for a lot of Atom-based CPUs due to instruction set/feature limitations a couple of years ago, so we don't want to leave these users on old and potentially vulnerable versions of Windows 10.

**Comment by monotron** [comment]
@cruxicheiros Secure Boot is largely a solved problem at this point - Ubuntu will offer to enrol its keys on install when it detects that Secure Boot is enabled. It might still be worth knowing how to disable it though, especially if we get into network booting

**Comment by markpastuszka** [comment]
@cruxicheiros If the key is legible on the COA sticker, you don't even need to do it as an upgrade install or use the tool - just enter the key into the installer when prompted and it should automatically select the version. The upgrade tool would only need to be used if the machine had a (clean) install of 7 on it and a missing/illegible product key.

Agreed on netbooks. I'd also consider installing Linux on systems such as HP Streams with 32GB eMMC storage as these often struggle with Windows updates given their small internal storage, again potentially risking users on vulnerable versions of 10.

**Comment by cruxicheiros** [comment]
@monotron thanks for the update!

**Comment by cruxicheiros** [comment]
Do you think we have to preload AV software? I think Windows Defender is good enough by now (Microsoft started taking it seriously sometime in the past so it no longer deserves its nasty reputation, and we can avoid some bloatware)

**Comment by markpastuszka** [comment]
I'd suggest sticking with Defender, a lot of free AVs can be a bit bloaty and would have negative impacts on performance.

**Comment by monotron** [comment]
@cruxicheiros Windows Defender is good enough imho, especially since a lot of free AV solutions deliver a relatively sub-par experience

**Comment by cruxicheiros** [comment]
**From Slack discussion**

For Ubuntu we can create a [custom Live CD](https://help.ubuntu.com/community/LiveCDCustomization) if needed.

We should consider what software will be used by the widest variety of users when we consider what to install, to avoid bloatware.

It's effectively already decided that all installs should come with free office software such as LibreOffice pre-installed. (Ubuntu may have this by default, MS Word only has a free trial, TextEdit is not a fully-fledged word processor)
